utils/zerovec/src/error.rs

```diff
@@ -10,11 +10,17 @@ use core::fmt;
 #[non_exhaustive]
 pub enum ZeroVecError {
     /// Attempted to parse a buffer into a slice of the given ULE type but its
-    /// length was not compatible
+    /// length was not compatible.
+    ///
+    /// Typically created using [`ZeroVecError::length()`].
     InvalidLength { ty: &'static str, len: usize },
-    /// The byte sequence provided for `ty` failed to parse correctly
+    /// The byte sequence provided for `ty` failed to parse correctly in the
+    /// given ULE type.
+    ///
+    /// Typically created using [`ZeroVecError::parse()`].
     ParseError { ty: &'static str },
-    /// The byte buffer was not in the appropriate format for VarZeroVec
+    /// The byte buffer was not in the appropriate format for VarZeroVec.
+    /// This error comes from `zerovec` as opposed to a ULE impl.
```

Comment:
I think if you have a `ZeroVecError` you should point out when an error *does not* come from `zerovec`. Also, by `zerovec`, do you mean the crate? Or do you mean `(Var)?Zero(Vec|Slice)`?

Can this error be split up?

Reply:
Is this better?

This is probably a good candidate for a Narrow Error Types treatment, but I'm just currently trying to improve the docs since #5002 didn't use the correct error variant